docs/program/apis/robot.md

```diff
@@ -7,11 +7,153 @@ description: "How to use the Robot API to monitor and manage your smart machines
+    creds = Credentials(
+        type='robot-location-secret',
+        payload='YOUR LOCATION SECRET')
+    opts = RobotClient.Options(
+        refresh_interval=0,
+        dial_options=DialOptions(credentials=creds)
+    )
+    return await RobotClient.at_address('ADDRESS FROM THE VIAM APP', opts)
+
+
+async def main():
+    # Make a RobotClient
+    robot_client = await connect()
+    print('Resources:')
+    print(robot.resource_names)
+    await robot_client.close()
+
+if __name__ == '__main__':
+    asyncio.run(main())
+```
+
+You can use this code to connect to your smart machine and instantiate a `RobotClient` that you can then use with the [robot API](#api).
+As an example, this code uses the instantiated `RobotClient` to return the {{< glossary_tooltip term_id="resource" text="resources" >}} currently configured.
+Remember to always close the connection (using `close()`) when done.
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go {class="line-numbers linkable-line-numbers"}
+package main
+
+import (
+  "context"
+
+  "github.com/edaniels/golog"
+  "go.viam.com/rdk/robot/client"
+  "go.viam.com/rdk/utils"
+  "go.viam.com/utils/rpc"
+)
+
+func main() {
+  logger := golog.NewDevelopmentLogger("client")
+  robot, err := client.New(
+      context.Background(),
+      "ADDRESS FROM THE VIAM APP",
+      logger,
+      client.WithDialOptions(rpc.WithCredentials(rpc.Credentials{
+          Type:    utils.CredentialsTypeRobotLocationSecret,
+          Payload: "YOUR LOCATION SECRET",
+      })),
+  )
+  if err != nil {
+      logger.Fatal(err)
+  }
+
+  defer robot.Close(context.Background())
+  logger.Info("Resources:")
+  logger.Info(robot.ResourceNames())
+}
+```
+
+You can use this code to connect to your smart machine and instantiate a `robot` client that you can then use with the [Robot API](#api).
+As an example, this code uses the instantiated `robot` client to return the {{< glossary_tooltip term_id="resource" text="resources" >}} currently configured.
+Remember to always close the connection (using `Close()`) when done.
+
+{{% /tab %}}
+{{< /tabs >}}
+
+Once you have instantiated the robot client, you can run any of the [robot API methods](#api) against the `robot` object to communicate with your smart machine.
+
+### Configure a timeout
+
+Because the robot API needs to be able to reliably connect to a deployed smart machine even over a weak or intermittent network, it supports a configurable timeout for connection and command execution.
+
+To configure a timeout when using the robot API:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Add a `timeout` to [`DialOptions`](https://python.viam.dev/autoapi/viam/rpc/dial/index.html#viam.rpc.dial.DialOptions):
+
+```python {class="line-numbers linkable-line-numbers"}
+# Add the timeout argument to DialOptions:
+dial_options = DialOptions(credentials=creds, timeout=10)
+```
+
+The default timeout in Python is 20 seconds.
+The example above shows a timeout of 10 seconds configured.
```

Comment:
Is there a maximum supported timeout value for Python?

Reply:
https://github.com/viamrobotics/rdk/pull/2886#discussion_r1387134594